yetanotherco/aligned_layer
# feat (batcher): finalize batch every n blocks

https://github.com/yetanotherco/aligned_layer/pull/182 needs to be merged first

Batcher now finalizes batch (uploads to s3 & submits task to Ethereum every N blocks or M tasks, whichever happens first.

You can configure batch_size_interval & block_interval on config.yaml

# To test

1. Start anvil, aggregator, operator & batcher (make sure to have env file with aws credentials)
2. Run `make batcher_send_dummy_task` 5 times
3. You should see batcher finalizing batch
4. Run `make batcher_send_dummy_task` twice (lambdaworks Merkle tree requires at least two leaves)
5. Run `make operator_get_eth` (just to make anvil generate new blocks)
6. Batcher should finalize batch
7. You'll see a call reverted error because Merkle tree is repeated, you can test both succesfully by restarting anvil between batches
